extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,14 +213,20 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (config['visibilitySpec']) {
+    const spec = config['visibilitySpec'];
+    if (spec) {
       if (!isVisibilitySpecValid(config)) {
         return;
       }
-
+      const self = this;
```

Comment:
From chrome debugger the context was pointing to global object, but console log gave me the right one.